api/src/main/java/org/openmrs/module/queue/api/QueueEntryService.java

```diff
@@ -48,6 +49,15 @@ public interface QueueEntryService {
 	 */
 	QueueEntry saveQueueEntry(@NotNull QueueEntry queueEntry);
 	
+	/**
+	 * Transitions a queue entry by ending one queue entry and creating a new queue entry that starts at
+	 * that time
+	 * 
+	 * @param queueEntryTransition the queueEntryTransition
+	 * @return the new QueueEntry that is created
+	 */
+	QueueEntry transitionQueueEntry(@NotNull QueueEntryTransition queueEntryTransition);
```

Comment:
Is this the right return type?  Should `QueueEntryTransition` have an additional property for the new queueEntry (which starts out null), and then this method would return the modified `QueueEntryTransition`?

Reply:
I think this is a lot cleaner that returning the `QueueEntryTransition`. It's also a good way of demarcating that the transition is over at this point.

Reply:
+1

Reply:
+1